aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -326,8 +326,16 @@ Tensor& reciprocal_out(Tensor& result, const Tensor& self) { return unary_op_imp
 Tensor reciprocal(const Tensor& self) { return unary_op_impl_float(self, reciprocal_stub); }
 Tensor& reciprocal_(Tensor& self) { return unary_op_impl_(self, at::reciprocal_out); }
 
-Tensor& rsqrt_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, rsqrt_stub); }
-Tensor rsqrt(const Tensor& self) { return unary_op_impl(self, at::rsqrt_out); }
+Tensor& rsqrt_out(Tensor& result, const Tensor& self) {
+  TORCH_CHECK(
+      self.scalar_type() != kBool, "rsqrt: bool dtype is not supported");
```

Comment:
This is a nice error message. So we don't promote boolean inputs to the default scalar type like we do with integers?

Reply:
Also, there are other dtypes that this is not listed as supporting in its OpInfo, like float16 on CPU. Why add a special error message for bool and not those types?

Reply:
Actually it is not needed any-more. Was having trouble with ASAN back then for input value `False` i.e. `0`. 
The reason I added it here back then was, int -> float promotion would cast bool to float and allow it.

Thanks to `__ubsan_ignore_float_divide_by_zero__`, there is no need for it.

Thanks for the catch. Will update and inform you.